src/components/features/Gallery/Gallery.js

```diff
@@ -13,8 +16,94 @@ import {
 import { faHeart } from '@fortawesome/free-regular-svg-icons';
 import ReactTooltip from 'react-tooltip';
 import StarsRating from '../StarsRating/StarsRating';
+import {
+  getAll,
+  getCompare,
+  toggleProductCompare,
+  toggleFavoriteProduct,
+} from '../../../redux/productsRedux';
+import { getGalleryCategories } from '../../../redux/categoriesRedux';
 
 const Gallery = () => {
+  const dispatch = useDispatch();
+  const allProducts = useSelector(getAll);
+  const galleryCategories = useSelector(getGalleryCategories);
+  const getCompareProducts = useSelector(getCompare);
+  const [activatePhotoFade, setActivatePhotoFade] = useState(false);
+  const [activateSliderFade, setActivateSliderFade] = useState(false);
+  const [activeCategory, setActiveCategory] = useState('topSeller');
+  const [productIndex, setProductIndex] = useState(0);
+
+  const categoryProducts = allProducts.filter(
+    item => item.galleryCategory === activeCategory
+  );
+  const [activeProduct, setActiveProduct] = useState(categoryProducts[0].id);
+  const handleCategoryChange = (e, newCategory) => {
+    e.preventDefault();
+    setActivatePhotoFade(true);
+    setActivateSliderFade(true);
+    setTimeout(() => {
+      setActiveCategory(newCategory);
+
+      setTimeout(() => {
```

Comment:
jaką rolę pełni ten timeout?

Reply:
Poprawione, to chyba pozostałość z momentu, w którym testowałem jak to będzie działać najlepiej. Usunąłem jeden timeout.